tests/test_package_generation.py

```diff
@@ -0,0 +1,61 @@
+"""Checks that the cookiecutter works."""
+
+import pathlib
+import subprocess
+from typing import Callable
+
+
+def test_package_generation(
+    tmp_path: pathlib.Path,
+    generate_package: Callable,
+) -> None:
+    """Test package generation."""
+    test_config = {
+        "github_username": "test-user",
+        "project_short_description": "description",
+        "project_name": "Cookiecutter Test",
+    }
+    generate_package(config=test_config, path=tmp_path)
+
+    # Check project directory exists
+    test_project_dir = tmp_path / "cookiecutter-test"
+    assert test_project_dir.exists()
+
+    # Check main files and directories inside
+    expected_files: list[str | pathlib.Path] = [
+        "README.md",
+        ".pre-commit-config.yaml",
+        "LICENSE.md",
+        "pyproject.toml",
+        "src",
+        pathlib.Path("src") / "cookiecutter_test",
+        pathlib.Path("src") / "cookiecutter_test" / "__init__.py",
+        "tests",
+        pathlib.Path(".github"),
+        pathlib.Path(".github") / "workflows",
+        "mkdocs.yml",
+        pathlib.Path("docs") / "index.md",
+        pathlib.Path("docs") / "api.md",
```

Comment:
Is this up-to-date?

Reply:
To my knowledge yes....?

Reply:
Presumably (hopefully!) the test would fail if it wasn't up to date

Reply:
Good point!